packages/framework/ai-collab/src/explicit-strategy/debugEvents.ts

```diff
@@ -0,0 +1,253 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { v4 as uuidv4 } from "uuid";
+
+import type { DebugEvent, EventFlowDebugEvent } from "../aiCollabApi.js";
+
+import type { TreeEdit } from "./agentEditTypes.js";
+
+/**
+ * A debug event for signaling the start of the ai-collab's core event loop.
+ * Which makes various calls to the LLM to eventually apply edits to the users SharedTree which
+ * accomplish the user's provided goal.
+ * @alpha
+ */
+export interface CoreEventLoopStartedDebugEvent extends EventFlowDebugEvent {
+	eventName: "CORE_EVENT_LOOP_STARTED";
+	eventFlowName: "CORE_EVENT_LOOP";
+	eventFlowStatus: "STARTED";
+}
+
+/**
+ * A debug event for signaling the end of the ai-collab's core event loop.
+ * There could be various reasons for the event loop to end, early exits and failures
+ * which should be captured in the status and failureReason fields.
+ * @alpha
+ */
+export interface CoreEventLoopCompletedDebugEvent extends EventFlowDebugEvent {
+	eventName: "CORE_EVENT_LOOP_COMPLETED";
+	eventFlowName: "CORE_EVENT_LOOP";
+	eventFlowStatus: "COMPLETED";
+	status: "success" | "failure";
+	failureReason?: string;
+	errorMessage?: string;
+}
+
+/**
+ * ----------------------------------------------------------------------------------
+ * Planning Prompt Debug events
+ * ----------------------------------------------------------------------------------
+ */
+
+/**
+ * A debug event marking the initiation of the flow for prompting an LLM to generate a plan for accomplishing the user's goal.
+ * @alpha
+ */
+export interface PlanningPromptStartedDebugEvent extends EventFlowDebugEvent {
+	eventName: "GENERATE_PLANNING_PROMPT_STARTED";
+	eventFlowName: "GENERATE_PLANNING_PROMPT";
+	eventFlowStatus: "STARTED";
+}
+
+/**
+ * A debug event marking the completion of the flow for prompting an LLM to generate a plan for accomplishing the user's goal.
+ * @alpha
+ */
+export interface PlanningPromptCompletedDebugEvent extends EventFlowDebugEvent {
```

Comment:
I wonder if we can get a bit more type safety (at the expense of a bit of readability) with something like:

```typescript
export interface PlanningPromptCompletedDebugEvent<TOutcome = "success" | "failure", TPlan = TOutcome extends "success" ? string : undefined> extends EventFlowDebugEvent {
	eventName: "GENERATE_PLANNING_PROMPT_COMPLETED";
	eventFlowName: "GENERATE_PLANNING_PROMPT";
	eventFlowStatus: "COMPLETED";
	requestOutcome: TOutcome;
	llmGeneratedPlan: TPlan;
}
```

So we guarantee that in the success case the plan is always a string, in the failure case it's always undefined. And similar idea for the other "completed" events.